packages/rfw/lib/src/dart/model.dart

```diff
@@ -439,6 +439,22 @@ class ConstructorCall extends BlobNode {
   String toString() => '$name($arguments)';
 }
 
+/// Representation of functions that return widgets in Remote Flutter library
+/// blobs.
+class FunctionDeclaration extends BlobNode {
+  /// Creates a [FunctionDeclaration] for a function that returns a widget.
+  const FunctionDeclaration(this.argumentName, this.constructorCall);
+
+  /// RFW functions can only take a single argument binded to this identifier.
+  final String argumentName;
+
+  /// RFW functions can only return a [ConstructorCall].
+  final ConstructorCall constructorCall;
```

Comment:
Should we support Switch here too?

Reply:
I don't think it is stricly needed. There could be two potential use cases of switch here:

#### A user wants to changed the passed value based on a given `args` or `state` entry value. 

For this particular case ,they can use a switch wrapping the builders

```
Builder(
  builder: switch args.down {
     true: (foo) => Text(...),
     false: (foo) => Container(...),
  }
);
```

#### A user wants to changed the passed value based on the scoped builder var
If I'm not wrong, you're suggesting something like this

```
Builder(
  builder:  (foo) => switch foo.down {
    case true:  (foo) => Text(...)
    case false:  (foo) => Text(...)
  }
);
```

but the same can be achieved by doing

Builder(
  builder:  (foo) => Foo(down: foo.down),
);

widget Foo = switch args.down {
 ...
}


Do you see any value in supporting switch here as well? (We can always add it later).

Reply:
Ended up implementing this.